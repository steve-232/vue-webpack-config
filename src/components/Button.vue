<template>
  <button @click="onClick">
    <slot />
  </button>
</template>

<script>
export default {
  props: {
    onClick: {
      type: Function,
      default: () => {},
    },
  },
};
</script>

<style lang="scss" scoped>
button {
  background-color: $white;
}
</style>
