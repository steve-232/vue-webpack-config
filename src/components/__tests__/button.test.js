import { mount } from '@vue/test-utils';
import Button from '../Button.vue';

describe('Button.vue', () => {
  it('should render custom text & on "click" invoke custom function & snapshot', () => {
    const customFun = jest.fn();
    const wrapper = mount(Button, {
      slots: {
        default: 'Custom text',
      },
      propsData: {
        onClick: customFun,
      },
    });

    // custom text
    expect(wrapper.text()).toBe('Custom text');

    // custom function
    wrapper.trigger('click');
    expect(customFun).toHaveBeenCalled();

    // snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
