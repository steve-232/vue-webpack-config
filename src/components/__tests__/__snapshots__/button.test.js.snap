// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue should render custom text & on "click" invoke custom function & snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
